### DISCUSSION TITLE
Clarification of Ref Topology

### DISCUSSION THREAD

**Discussion by SMorettini** [nasa/fprime#1805] [Q&A]
I notice the following line in the Topology of Ref and I don't understand it:

https://github.com/nasa/fprime/blob/91c578ee37c01ba92e556dbca0287d82f9f18b95/Ref/Top/topology.fpp#L121

Why the `rateGroup3Comp` is calling the `Sched` port of `blockDrv`? It seems like a loop. Moreover the `Sched` port handler doesn't contain anything.

Part of code from Ref topology:
```
      # Block driver
      blockDrv.CycleOut -> rateGroupDriverComp.CycleIn

      ...

      # Rate group 3
      rateGroupDriverComp.CycleOut[Ports_RateGroups.rateGroup3] -> rateGroup3Comp.CycleIn
      ...
      rateGroup3Comp.RateGroupMemberOut[2] -> blockDrv.Sched
```

**Comment by timcanham** [comment]
I can see how it would be confusing. `blockDrv` was meant to be a demonstration of a trivial device driver. It is supposed to do two things:

1\) Supply a "clock" tick to drive rate groups. In a real-world example, this could be tied to a hardware timer. We emulate this by calling the "interrupt service routine" manually from our `main()`: 

https://github.com/nasa/fprime/blob/master/Ref/Top/Main.cpp#L28

2\) Accept "packets" for transmission out and send any "received" packets to users. In this case, to keep it trivial, we just loop back the packets.

We typically use the `Sched` port to push out driver statistics periodically. This is to avoid overloading the telemetry system by pushing statistics faster, say when we get each packet in a high-rate system. We should add an example of this to the code so it doesn't confuse readers, and add some more comments to explain the pattern. I added #1807 to update the example.
